### ISSUE TITLE
Last updated on time / author is always same for every page?

### ISSUE THREAD

**Issue by Jason-2020** [facebook/docusaurus#2798]
## 📚 Documentation

`enableUpdateTime` option enables _Last updated on DD/MM/YYYY_ feature. However this value is same for every page. I guess this was working fine in version 1 _(it's broken when we started using github actions tho..)_ But it's not working as expected in v2. Is this bug or limitation?
![image](https://user-images.githubusercontent.com/58852063/82719167-e1177280-9c6d-11ea-8c92-7b2a88e8ddd6.png)

related: https://github.com/facebook/docusaurus/issues/1015
### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

**Comment by yangshun** [comment]
Any live repro example? If you're talking about versioned pages that's
expected.

**Comment by yangshun** [comment]
If you're looking at the pages under https://v2.docusaurus.io/docs/, it's pointing to the latest released/stable version of the docs, and that's created when a new documentation version is cut, which was few days ago when we released alpha.55. If you see the trunk version of the docs under `docs/next/` (e.g. https://v2.docusaurus.io/docs/next/using-plugins), you'll see that the author is different.

**Comment by yangshun** [comment]
Closing this out as it's not a bug.

**Comment by Jason-2020** [comment]
Hi @yangshun 

> If you're looking at the pages under https://v2.docusaurus.io/docs/, it's pointing to the latest released/stable version of the docs, and that's created when a new documentation version is cut, which was few days ago when we released alpha.55. If you see the trunk version of the docs under `docs/next/` (e.g. https://v2.docusaurus.io/docs/next/using-plugins), you'll see that the author is different.

Thanks for the explanation. In our case we don't use versioned pages.
Please take a look at our documentation [website](https://docs.erxes.io/overview/getting-started/). [Here](https://github.com/erxes/erxes/tree/develop/docs) is the repo.

**Comment by Jason-2020** [comment]
pinging 1 more time @yangshun

**Comment by yangshun** [comment]
The last updated date look at Git history of a file, so I suspect it could be the environment of your build step. When you clone the repo on your deployment environment do you happen to be doing just a shallow clone? A preferably long history is needed for this to work well.

**Comment by Jason-2020** [comment]
We're using github actions to deploy to github pages directly. [Here](https://github.com/erxes/erxes/blob/develop/.github/workflows/documentation.yaml) is our workflow file.

Main part is:

```yml
- name: Publish
      if: github.event_name == 'push'
      run: |
        cd docs
        git config --global user.email "$DOC_USERNAME@users.noreply.github.com"
        git config --global user.name "$DOC_USERNAME"
        echo "machine github.com login $DOC_USERNAME password $DOC_TOKEN" > ~/.netrc
        GIT_USER=$DOC_USERNAME CURRENT_BRANCH=develop yarn deploy
```

**Comment by yangshun** [comment]
Could you try creating a custom build on your local machine and see if the timestamps work properly? If so then it's something with GitHub actions environment. Please let us know :)

**Comment by Jason-2020** [comment]
I confirmed that `showLastUpdateAuthor` and `showLastUpdateTime` feature is working as expected on local builds. I guess I'll tune github checkout actions' settings then. Thank you :)

**Comment by slorber** [comment]
Thanks @Jason-2020 

Do you think we should include this in our GH pages deployment doc?

(it seems we are still on actions/checkout@v1, might be good to update the CI config a bit)

I'm not very familiar with Github CI but would merge a PR to improve/update this part of the doc

**Comment by dfblhmm** [comment]
This Bug is not resolved util v3.1.0
I only can use the [markdown.parseFrontMatter] to calculate last_update automatically

``` js
import { statSync } from "fs";

export default {
  markdown: {
    parseFrontMatter: async (params) => {
      const { filePath, defaultParseFrontMatter } = params;

      const { frontMatter, ...rest } = await defaultParseFrontMatter(params);

      const { mtime } = statSync(filePath);
      const lastUpdateTime = new Date(mtime).toLocaleString();

      return {
        ...rest,
        frontMatter: {
          ...frontMatter,
          last_update: {
            date: lastUpdateTime
          }
        }
      };
    }
  }
};
```
